pkg/build/android.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/google/syzkaller/pkg/osutil"
+)
+
+const (
+	kernelConfig = "common/build.config.gki_kasan.x86_64"
+	moduleConfig = "common-modules/virtual-device/build.config.virtual_device_kasan.x86_64"
+)
+
+type android struct{}
+
+func (a android) runBuild(kernelDir, buildConfig string) error {
+	cmd := osutil.Command("build/build.sh")
+	cmd.Dir = kernelDir
+	cmd.Env = append(cmd.Env, "DIST_DIR=out/dist", fmt.Sprintf("BUILD_CONFIG=%s", buildConfig))
+
+	_, err := osutil.Run(time.Hour, cmd)
+	return err
+}
+
+func (a android) build(params Params) (ImageDetails, error) {
+	var details ImageDetails
+
+	if params.CmdlineFile != "" {
+		return details, fmt.Errorf("cmdline file is not supported for android cuttlefish images")
+	}
+	if params.SysctlFile != "" {
+		return details, fmt.Errorf("sysctl file is not supported for android cuttlefish images")
+	}
+
+	if err := a.runBuild(params.KernelDir, kernelConfig); err != nil {
+		return details, fmt.Errorf("failed to build kernel: %s", err)
+	}
+	if err := a.runBuild(params.KernelDir, moduleConfig); err != nil {
+		return details, fmt.Errorf("failed to build modules: %s", err)
+	}
+
+	buildOutDir := filepath.Join(params.KernelDir, "out/dist")
```

Comment:
Strictly speaking, the proper way to do this is:
`filepath.Join(params.KernelDir, "out", "dist")`
(e.g. windows uses '\' as path separator)